pkg/controller/migplan/validation.go

```diff
@@ -759,6 +773,157 @@ func (r ReconcileMigPlan) validatePvSelections(plan *migapi.MigPlan) error {
 	return nil
 }
 
+// Validate proxy secrets. Should only exist 1 or none
+func (r ReconcileMigPlan) validateRegistryProxySecrets(plan *migapi.MigPlan) error {
+	err := r.validateSourceRegistryProxySecret(plan)
+	if err != nil {
+		return err
+	}
+	return r.validateDestinationRegistryProxySecret(plan)
+}
+
+// Validate registry proxy secret ensuring it has the proper keys
+func (r *ReconcileMigPlan) validateRegistryProxySecret(secret *kapi.Secret) bool {
+	fields := []string{
+		settings.HttpProxy,
+		settings.HttpsProxy,
+		settings.NoProxy,
+	}
+	for _, key := range fields {
+		if _, found := secret.Data[key]; found {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Validate source proxy secret
+func (r ReconcileMigPlan) validateSourceRegistryProxySecret(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended, InvalidSourceClusterRef, SourceClusterNotReady) {
+		return nil
+	}
+
+	list := kapi.SecretList{}
+	selector := k8sLabels.SelectorFromSet(map[string]string{
+		"migration-proxy-config": "true",
+	})
+
+	// Source cluster proxy secret validation
+	srcCluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if srcCluster == nil || !srcCluster.Status.IsReady() {
+		return nil
+	}
+
+	srcClient, err := srcCluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	err = srcClient.List(
+		context.TODO(),
+		&k8sclient.ListOptions{
+			Namespace:     migapi.VeleroNamespace,
+			LabelSelector: selector,
+		},
+		&list,
+	)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	if len(list.Items) == 0 {
+		// No proxy secret is valid configuration
+		return nil
+	}
+	if len(list.Items) > 1 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     SourceClusterProxySecretMisconfigured,
+			Status:   True,
+			Reason:   Conflict,
+			Category: Critical,
+			Message:  SourceClusterProxySecretMisconfiguredMessage,
+		})
+		return nil
+	}
+	if !r.validateRegistryProxySecret(&list.Items[0]) {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     SourceClusterProxySecretMisconfigured,
+			Status:   True,
+			Reason:   KeyNotFound,
+			Category: Critical,
+			Message:  SourceClusterProxySecretMisconfiguredMessage,
+		})
+	}
+	return nil
+}
+
+// Validate destination proxy secret
+func (r ReconcileMigPlan) validateDestinationRegistryProxySecret(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended, InvalidDestinationClusterRef, DestinationClusterNotReady) {
+		return nil
+	}
+
+	list := kapi.SecretList{}
+	selector := k8sLabels.SelectorFromSet(map[string]string{
+		"migration-proxy-config": "true",
+	})
+
+	// Destination cluster proxy secret validation
+	destCluster, err := plan.GetDestinationCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if destCluster == nil || !destCluster.Status.IsReady() {
```

Comment:
The IsReady() here is fine but a little redundant with the HasAnyCondition() check on `DestinationClusterNotReady` above.